test/PackageVersionsTest/E2EInstallerTest.php

```diff
@@ -184,6 +186,60 @@ public function testRemovingPluginWithNoDevDoesNotAttemptToGenerateVersions() :
         );
     }
 
+    public function testOnReadonlyFilesystemDoesNotGenerateClasses() : void
+    {
+        $this->createPackageVersionsArtifact();
+        $this->createArtifact();
+
+        $this->writeComposerJsonFile(
+            [
+                'name'         => 'package-versions/e2e-local',
+                'require-dev'  => ['ocramius/package-versions' => '1.0.0'],
+                'repositories' => [
+                    ['packagist' => false],
+                    [
+                        'type' => 'artifact',
+                        'url' => $this->tempArtifact,
+                    ],
+                ],
+            ],
+            $this->tempLocalComposerHome
+        );
+
+        $this->execComposerInDir('install', $this->tempLocalComposerHome);
+
+        unlink(
```

Comment:
Waaait... `Versions.php` should exist at all times (it's part of the sources of this package).

I'd rather expect it to not be writable, but to certainly exist